app/Http/Controllers/ProfileController.php

```diff
@@ -127,7 +128,23 @@ public function update(Request $request)
 		}
 
 		$profile = \Auth::user()->profile;
-		$profile->update($request->all());
+
+		if (\Auth::user()->profile_type == 'App\Participant') {
+			$profile->update($request->all());
+		} else {
+			$profile->update($request->except('skills'));
+
+			// Update skills
+			$skills = $request->input('skills') ?: []; // this is an array with ids of existing skills (as strings!) and string tags of new skills
```

Comment:
(See previous comment )

Could you use ?? which is the null coalescing operator and this is exactly what it is for. If for some reason the $request->input('skills') returns an empty string it won't default to [] which is not preferable I think.